pkg/auth/handler/sso/sso.go

```diff
@@ -126,7 +126,14 @@ func (h respHandler) handleLogin(
 		err = nil
 	}
 
-	// TODO: handle auto link user
+	if valid := h.PasswordAuthProvider.IsAuthDataMatching(oauthAuthInfo.ProviderAuthData); valid {
```

Comment:
Seems it does not restrict the auth record key to `email` 🤔

Reply:
I am thinking since `oauthAuthInfo.ProviderAuthData` contains `email` for now, and if `IsAuthDataMatching` is true, it means `AuthRecordKeys` contains `["email"]`. And in the future, if skygear next support hook mechanism when process `AuthData`, it will return correct `authData` by the hook function.

Do you think such idea correct? Or I miss anything.

Reply:
I added some comments, see if it is helpful. `oauthAuthInfo.ProviderAuthData` is the same structure of `authData`, like `{email: "john.doe@example.com"}`